### PR TITLE
Ux buttons btn unstyled, alignment, text

### PR DESCRIPTION
**Pull request by bisht-richa**
## What does this PR change?

- A new button style was added: _btn-tertiary_. 
- Replaced _btn-link_ with _btn-tertiary_
- Fixed button labels consistency.
- Corrected button placement for better usability and uniformity across the interface.

**add description**

## GUI diff

No difference.

Before:
Screenshot 2024-11-18 at 13 56 58

Screenshot 2024-11-19 at 18 29 13
Screenshot 2024-12-05 at 16 57 59
Screenshot 2025-01-08 at 13 06 33

After:
Screenshot 2025-01-08 at 13 08 11

Screenshot 2024-12-16 at 15 08 33
Screenshot 2025-01-08 at 13 06 19

**btn-tertiary**
Screenshot 2025-01-17 at 13 05 10
Screenshot 2025-01-17 at 13 05 17

- [x] **DONE**

## Documentation
- No documentation needed

- [x] **DONE**

## Test coverage
ℹ️ If a major new functionality is added, it is **strongly recommended** that tests for the new functionality are added to the Cucumber test suite

- No tests: already covered

- [x] **DONE**

## Links

Fixes: [#26167](https://github.com/SUSE/spacewalk/issues/26167)
Port(s): # **add downstream PR(s), if any**

- [x] **DONE**

## Changelogs

Make sure the changelogs entries you are adding are compliant with https://github.com/uyuni-project/uyuni/wiki/Contributing#changelogs and https://github.com/uyuni-project/uyuni/wiki/Contributing#uyuni-projectuyuni-repository

If you don't need a changelog check, please mark this checkbox:

- [ ] No changelog needed

If you uncheck the checkbox after the PR is created, you will need to re-run `changelog_test` (see below)

## Re-run a test

If you need to re-run a test, please mark the related checkbox, it will be unchecked automatically once it has re-run:

- [ ] Re-run test "changelog_test"
- [ ] Re-run test "backend_unittests_pgsql"
- [ ] Re-run test "java_pgsql_tests"
- [ ] Re-run test "schema_migration_test_pgsql"
- [ ] Re-run test "susemanager_unittests"
- [ ] Re-run test "javascript_lint"
- [ ] Re-run test "spacecmd_unittests"

# Before you merge

Check [How to branch and merge properly](https://github.com/uyuni-project/uyuni/wiki/How-to-branch-and-merge-properly)!

### CODE REVIEW

File: web/html/src/components/table/SearchPanel.tsx [hunk 1]
```diff
@@ -60,18 +60,7 @@ export function SearchPanel(props: SearchPanelProps) {
...
-          <span>
-            {t("({selectedCount} selected)", { selectedCount: props.selectedCount })}&nbsp;
-            <button className="btn-link" onClick={props.onClear}>
-              {t("Clear")}
-            </button>
-            &nbsp;/&nbsp;
-          </span>
-        )}
-        {props.selectable && (
-          <button className="btn-link" onClick={props.onSelectAll}>
```
[the remark this answers is not in the document]
Comment (bisht-richa) [reply web/html/src/components/table/SearchPanel.tsx:72]: For me, it's all about seeing what I'm doing. When working with a huge list, I want to select only what's in front of me. It just feels safer, especially if I'm deleting stuff or making changes I can't undo.  I don't want to accidentally mess with things I can't see.

But we can discuss it with everyone and see what use cases we have or we can improve this interaction.

File: testsuite/features/secondary/minssh_action_chain.feature [hunk 2]
```diff
@@ -187,7 +187,7 @@ Feature: Salt SSH action chain
   Scenario: Delete the action chain for SSH minion
     When I follow the left menu "Schedule > Action Chains"
     And I follow "new action chain"
-    And I follow "delete action chain" in the content area
+    And I follow "delete action" in the content area
```
Comment (srbarrios) [inline testsuite/features/secondary/minssh_action_chain.feature:190]: I'm not sure this is correct, we name "action chain" in the menu, and we name "action chain" when we are creating it, why we name it only "action" when we are deleting it? That's not consistent.

Also, the title of this scenario is not consistent. Deleting an action should be different than deleting a chain of actions.

**Review by srbarrios — DISMISSED** [review]
Testsuite code changes LGTM.

File: java/code/webapp/WEB-INF/pages/admin/users/activelist.jsp [hunk 3]
```diff
@@ -20,8 +20,9 @@
 <rl:listset name="userListSet">
 <rhn:csrf />
 <rhn:submitted />
-
-
+     <rl:csv dataset="pageList"
```
Comment (Etheryte) [inline java/code/webapp/WEB-INF/pages/admin/users/activelist.jsp:23]: Weird whitespace?

File: web/html/src/components/buttons.tsx [hunk 4]
```diff
@@ -159,17 +158,23 @@ export type ButtonProps = BaseProps & {
...
         type="button"
         title={this.props.title}
-        className={"btn " + (this.props.className ?? "")}
+        className={cssClasses}
         onClick={this.props.handler}
         disabled={this.props.disabled}
       >
-        {this.renderIcon()}
-        {text}
+        {
```
Comment (Etheryte) [inline web/html/src/components/buttons.tsx:172]: The brackets and the fragment are superfluous here, what was present before works fine:

```
{this.renderIcon()}
{text}
```

File: web/html/src/components/buttons.tsx [hunk 5]
```diff
@@ -203,15 +210,20 @@ export class LinkButton extends _ButtonBase<LinkProps> {
...
         title={this.props.title}
-        className={"btn " + this.props.className}
+        className={cssClasses}
         href={this.props.href}
         onClick={this.props.handler}
         {...targetProps}
       >
-        {this.renderIcon()}
-        {text}
+        {
```
Comment (Etheryte) [inline web/html/src/components/buttons.tsx:221]: Same comment as earlier in this file.

File: web/html/src/components/buttons.tsx [hunk 6]
```diff
@@ -203,15 +210,20 @@ export class LinkButton extends _ButtonBase<LinkProps> {
           };
     return (
       <a
+        role="button"
```
Comment (Etheryte) [inline web/html/src/components/buttons.tsx:213]: I think this is not correct from an accessibility point of view. At the end of the day, this component works as a link even if it looks like a button, in the sense that if the user clicks it, they will be taken to another page. Setting the role to button contradicts that, so we should not set this value here.

**Review by Etheryte — COMMENTED** [review]
The frontend changes look good, a few small nitpicks, but nothing important.  

I don't have any input on the translation changes and such as it's hard to follow them without the context.

**Comment by Etheryte** [comment]
It is currently hard to figure out what of the previous discussions are still relevant and what's done, could you either reply or resolve existing comments as necessary @bisht-richa?

File: java/code/src/com/suse/manager/webui/templates/groups/formulas.jade [hunk 7]
```diff
@@ -1,9 +1,9 @@
 .spacewalk-toolbar-h1
   .spacewalk-toolbar
-    a(href="/rhn/groups/Delete.do?sgid=#{groupId}")
+    a(href="/rhn/groups/Delete.do?sgid=#{groupId}" class="btn btn-danger")
```
Comment (Etheryte) [inline java/code/src/com/suse/manager/webui/templates/groups/formulas.jade:3]: Here and elsewhere in Jade files, it would be nice to use a consistent style for class names. For example we have `a(class="btn btn-default")`, but also `i.fa.fa-trash-o` in the same file. It would be preferable if we kept a consistent style and used one or the other. This is not an important detail though.

File: java/code/webapp/WEB-INF/pages/admin/multiorg/orgtrust.jsp [hunk 8]
```diff
@@ -25,7 +25,14 @@
 <div class="page-summary" style="padding-top: 10px;">
 <p><bean:message key="org.trust.summary" arg0="${fn:escapeXml(org.name)}"/></p>
 </div>
+
 <rl:listset name="trustedOrgs">
+   <div class="spacewalk-section-toolbar">
+      <div class="action-button-wrapper">
+      <rhn:submitted/>
+      <input class="btn btn-default" type="submit" name ="confirm" value="${rhn:localize('org.trust.modify')}" />
```
Comment (Etheryte) [inline java/code/webapp/WEB-INF/pages/admin/multiorg/orgtrust.jsp:33]:
```suggestion
      <input class="btn btn-default" type="submit" name="confirm" value="${rhn:localize('org.trust.modify')}" />
```

File: web/html/src/manager/content-management/list-filters/appstreams/appstreams.tsx [hunk 9]
```diff
@@ -31,10 +32,12 @@ export default function AppStreams({ matcher }) {
...
           <div className="col-md-offset-3 offset-md-3 col-md-6">
-            <button className="btn-link" onClick={enableBrowse}>
-              {isLoading ? <i className="fa fa-refresh fa-spin fa-fw" /> : <i className="fa fa-search fa-fw" />}
-              Browse available modules
-            </button>
+            <Button
+              className="btn-tertiary"
+              handler={enableBrowse}
+              icon={isLoading ? "fa-refresh fa-spin fa-fw" : "fa-search fa-fw"}
+              text="Browse available modules"
```
Comment (Etheryte) [inline web/html/src/manager/content-management/list-filters/appstreams/appstreams.tsx:39]:
```suggestion
              text={t("Browse available modules")}
```

File: web/html/src/manager/systems/all-list.tsx [hunk 10]
```diff
@@ -55,6 +55,20 @@ export function AllSystems(props: Props) {
...
+        <div className="action-button-wrapper">
+          <a
+            role="button"
+            title="Download CSV"
+            href="/rhn/manager/systems/csv/all"
+            className="btn btn-default"
+            data-senna-off="true"
+          >
+            <IconTag type="item-download-csv" />
+            Download CSV
```
Comment (Etheryte) [inline web/html/src/manager/systems/all-list.tsx:68]:
```suggestion
            {t("Download CSV")}
```

File: web/html/src/manager/systems/virtual-list.tsx [hunk 11]
```diff
@@ -44,7 +44,19 @@ export function VirtualSystems(props: Props) {
...
+      <div className="spacewalk-section-toolbar">
+        <div className="action-button-wrapper">
+          <a
+            href="/rhn/manager/systems/csv/virtualSystems"
+            title="Download CSV"
+            className="btn btn-default"
+            data-senna-off="true"
+          >
+            <IconTag type="item-download-csv" />
+            Download CSV
```
Comment (Etheryte) [inline web/html/src/manager/systems/virtual-list.tsx:56]:
```suggestion
            {t("Download CSV")}
```

**Review by Etheryte — APPROVED** [review]
The frontend changes look good. 👍

Pinging @parlt91 regarding the older discussion points, I couldn't figure out if everything is addressed or not.

File: java/code/src/com/suse/manager/webui/templates/groups/formulas.jade (hunk 7 above)
Comment (bisht-richa) [reply java/code/src/com/suse/manager/webui/templates/groups/formulas.jade:3]: If you see it in any specific file please let me know I can not find this use case.

Comment (bisht-richa) [reply java/code/src/com/suse/manager/webui/templates/groups/formulas.jade:3]: I tried to fix this in most places, but if we find any inconsistencies, let's fix them in a new PR.

Comment (Etheryte) [reply java/code/src/com/suse/manager/webui/templates/groups/formulas.jade:3]: This issue is present in pretty much all touched Jade files in this PR. Allow me to rephrase so what I mean is clearer.  

In Jade files, you can create a div with `<div class="foo">` in two different ways: either as `.div` or as `div(class="foo")`. Existing code uses the first option throughout, so e.g. `i.fa.fa-trash-o`, but your additions use the other way, so e.g. `a(class="btn btn-danger")`. In this context there is no good reason to mix and match them, it would be preferable to use a consistent style throughout.

File: web/html/src/branding/css/base/components/buttons.suma.scss [hunk 12]
```diff
@@ -31,7 +31,7 @@
   }
 
   // Branding hover effect
-  &:not(:disabled):not(.disabled):not(.text-muted):hover::before {
+  &:not(:disabled):not(.disabled):not(.text-muted):hover::after {
```
Comment (Etheryte) [inline web/html/src/branding/css/base/components/buttons.suma.scss:34]: I would recommend adding a comment here that it has to be `after` and why, otherwise this bug might creep in again at a later date. 🙂

File: java/code/src/com/suse/manager/webui/templates/groups/formulas.jade (hunk 7 above)
Comment (bisht-richa) [reply java/code/src/com/suse/manager/webui/templates/groups/formulas.jade:3]: Okay got it, my mistake. I will fix this.

**Comment by bisht-richa** [comment]
Still relevant — will update after review.

File: web/html/src/components/hub/AddTokenButton.tsx [hunk 13]
```diff
@@ -86,22 +86,24 @@ export class AddTokenButton extends React.Component<Props, State> {
             title={t("Add a new access token")}
             className="btn-primary"
             items={[
-              <LinkButton
+              // eslint-disable-next-line jsx-a11y/anchor-is-valid
```
Comment (Etheryte) [inline web/html/src/components/hub/AddTokenButton.tsx:89]: We do not want to introduce new instances of this. `<a href="#" ...>` is bad practice and we only have it elsewhere because we haven't had the time to refactor old instances out. A simple mental model to use is that if clicking on it navigates you somewhere, it should be a link, if it performs some other action, it should be a button element. If we need the style to be different, we can use `<button className="is-plain>` or other similar options.

**Review by Etheryte — APPROVED** [review]
Looks good otherwise, but please fix the `<a href="#">` usage.

File: web/html/src/components/hub/AddTokenButton.tsx (hunk 13 above)
Comment (bisht-richa) [reply web/html/src/components/hub/AddTokenButton.tsx:89]: I have fixed this in another PR for the dropdowns.

**Comment by bisht-richa** [comment]
> Looks good otherwise, but please fix the `<a href="#">` usage.

I’ve fixed the changes you mentioned in a separate PR, along with some additional updates for the dropdowns.

File: web/html/src/components/hub/AddTokenButton.tsx (hunk 13 above)
Comment (Etheryte) [reply web/html/src/components/hub/AddTokenButton.tsx:89]: Nice, I must have missed that. In that case, all good, thanks a lot for the large amount of time you've put into this. 🙂
